drivers/imu/adis.c

```diff
@@ -0,0 +1,2075 @@
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include "adis.h"
+#include "no_os_delay.h"
+#include "no_os_gpio.h"
+#include "no_os_util.h"
+#include "no_os_print_log.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+#define ADIS_PAGE_SIZE			0x80
+#define ADIS_REG_PAGE_ID		0x00
+#define ADIS_WRITE_REG(reg)		((NO_OS_BIT(7) | (reg)))
+#define ADIS_READ_REG(reg)		((reg) & NO_OS_GENMASK(6,0))
+#define ADIS_4_BYTES_SIZE		4
+#define ADIS_2_BYTES_SIZE		2
+#define ADIS_1_BYTE_SIZE		1
+#define ADIS_16_BIT_BURST_SIZE		0
+#define ADIS_32_BIT_BURST_SIZE		1
+#define ADIS_MSG_SIZE_16_BIT_BURST 	20 /* in bytes */
+#define ADIS_MSG_SIZE_32_BIT_BURST 	32 /* in bytes */
+#define ADIS_CHECKSUM_SIZE		2  /* in bytes */
+#define ADIS_READ_BURST_DATA_CMD_SIZE	2  /* in bytes */
+#define ADIS_READ_BURST_DATA_CMD_MSB	0x68
+#define ADIS_READ_BURST_DATA_CMD_LSB	0x00
+#define ADIS_SIGN_BIT_POS		15
+#define ADIS_DIAG_IDX_16_BIT_BURST	0
+#define ADIS_XGYRO_IDX_16_BIT_BURST	2
+#define ADIS_YGYRO_IDX_16_BIT_BURST	4
+#define ADIS_ZGYRO_IDX_16_BIT_BURST	6
+#define ADIS_XACCL_IDX_16_BIT_BURST	8
+#define ADIS_YACCL_IDX_16_BIT_BURST	10
+#define ADIS_ZACCL_IDX_16_BIT_BURST	12
+#define ADIS_TEMP_IDX_16_BIT_BURST	14
+#define ADIS_CNT_IDX_16_BIT_BURST	16
+#define ADIS_DIAG_IDX_32_BIT_BURST	0
+#define ADIS_XGYRO_IDX_32_BIT_BURST	2
+#define ADIS_YGYRO_IDX_32_BIT_BURST	6
+#define ADIS_ZGYRO_IDX_32_BIT_BURST	10
+#define ADIS_XACCL_IDX_32_BIT_BURST	14
+#define ADIS_YACCL_IDX_32_BIT_BURST	18
+#define ADIS_ZACCL_IDX_32_BIT_BURST	22
+#define ADIS_TEMP_IDX_32_BIT_BURST	26
+#define ADIS_CNT_IDX_32_BIT_BURST	28
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Initialize adis device.
+ * @param adis - The adis device.
+ * @param data - Initialization data.
+ * @return 0 in case of success, error code otherwise.
+ */
+int adis_init(struct adis_dev **adis, const struct adis_data *data)
+{
+	struct adis_dev *dev;
+	int ret;
+
+	dev = (struct adis_dev *)calloc(1, sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+
+	ret = no_os_spi_init(&dev->spi_desc, data->adis_ip->spi_init);
+	if (ret)
+		goto error_spi;
+
+	if(data->has_paging)
+		dev->current_page = -1;
+	else
+		dev->current_page = 0;
+
+	ret = no_os_gpio_get_optional(&dev->gpio_reset, data->adis_ip->gpio_reset);
+	if (ret) {
+		pr_warning("No reset pin found \n");
+		goto error_gpio;
```

Comment:
Should this be an error?

Reply:
no, it should be just a warning and should continue with initialization, will modify, thanks!